windows/src/desktop/kmshell/install/Keyman.Configuration.System.TIPMaintenance.pas

```diff
@@ -177,6 +178,13 @@ class function TTIPMaintenance.DoRegister(const KeyboardID,
     Result := False;
 end;
 
+class function TTIPMaintenance.GetFirstLanguage(Keyboard: IKeymanKeyboardFile): string;
+begin
+  if Keyboard.Languages.Count > 0
+    then Result := Keyboard.Languages[0].BCP47Code
+    else Result := TLanguageCodeUtils.TranslateWindowsLanguagesToBCP47(HKLToLanguageID(GetDefaultHKL));
```

Comment:
Is this returning the default locale language tag? (which could be completely separate from the Keyboard.Languages[] list)

Reply:
Yes, `GetDefaultHKL` returns the user's default keyboard handle, and that's translated to a `LANGID` and from there into a BCP 47 tag. The ultimate fallback!